vm/src/builtins/bytes.rs

```diff
@@ -531,6 +531,13 @@ impl Comparable for PyBytes {
     ) -> PyResult<PyComparisonValue> {
         Ok(if let Some(res) = op.identical_optimization(zelf, other) {
             res.into()
+        } else if other.isinstance(&vm.ctx.types.memoryview_type) {
+            return Err(vm.new_type_error(format!(
+                "'{}' not supported between instances of '{}' and '{}'",
+                op.operator_token(),
+                zelf.class().name,
+                other.class().name
+            )));
         } else {
             zelf.inner.cmp(other, op, vm)
         })
```

Comment:
The CPython's logic is very strange here, memoryview could do == or != here with bytes, but raise type error when doing <, >, <=, >=. and bytearray just accept everything.
This code does not match the cpython's logic for the following compare:
```
bytes (==,!=) memoryview
memoryview (<,<=,>,>=)  bytearray
```